--- Discord/discord.desktop     2017-01-04 20:08:18.000000000 -0200
+++ /tmp/discord.desktop        2017-01-11 03:21:53.786174425 -0200
@@ -3,7 +3,7 @@
 StartupWMClass=discord
 Comment=All-in-one voice and text chat for gamers that's free, secure, and works on both your desktop and phone.
 GenericName=Internet Messenger
-Exec=/usr/share/discord/Discord
+Exec=/usr/bin/discord
 Icon=discord
 Type=Application
 StartupNotify=true

