From 8541e96a136cfbe41a4063fdb75bea35a287f079 Mon Sep 17 00:00:00 2001
From: Rodrigo Saboya <saboya@users.noreply.github.com>
Date: Sat, 5 May 2018 15:27:21 -0300
Subject: [PATCH] Adding env var for opting into 24-bit Vulkan formats.

---
 src/dxgi/dxgi_format.cpp | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/dxgi/dxgi_format.cpp b/src/dxgi/dxgi_format.cpp
index b18e1ce..527b2fb 100644
--- a/src/dxgi/dxgi_format.cpp
+++ b/src/dxgi/dxgi_format.cpp
@@ -3,7 +3,8 @@
 #include <array>
 
 namespace dxvk {
-  
+  bool use24bitFormat = env::getEnvVar(L"DXVK_USE_VULKAN_24BIT_FORMAT") == "1";
+
   std::array<DXGI_VK_FORMAT_MAPPING, 133> g_dxgiFormats = {{
     // DXGI_FORMAT_UNKNOWN
     { },
@@ -216,21 +217,21 @@ namespace dxvk {
       VK_IMAGE_ASPECT_COLOR_BIT },
     // DXGI_FORMAT_R24G8_TYPELESS
     { VK_FORMAT_UNDEFINED,
-      VK_FORMAT_D32_SFLOAT_S8_UINT,
+      use24bitFormat ? VK_FORMAT_D24_UNORM_S8_UINT : VK_FORMAT_D32_SFLOAT_S8_UINT,
       VK_FORMAT_UNDEFINED },
     // DXGI_FORMAT_D24_UNORM_S8_UINT
     { VK_FORMAT_UNDEFINED,
-      VK_FORMAT_D32_SFLOAT_S8_UINT,
+      use24bitFormat ? VK_FORMAT_D24_UNORM_S8_UINT : VK_FORMAT_D32_SFLOAT_S8_UINT,
       VK_FORMAT_UNDEFINED,
       0, VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT },
     // DXGI_FORMAT_R24_UNORM_X8_TYPELESS
     { VK_FORMAT_UNDEFINED,
-      VK_FORMAT_D32_SFLOAT_S8_UINT,
+      use24bitFormat ? VK_FORMAT_D24_UNORM_S8_UINT : VK_FORMAT_D32_SFLOAT_S8_UINT,
       VK_FORMAT_UNDEFINED,
       0, VK_IMAGE_ASPECT_DEPTH_BIT },
     // DXGI_FORMAT_X24_TYPELESS_G8_UINT
     { VK_FORMAT_UNDEFINED,
-      VK_FORMAT_D32_SFLOAT_S8_UINT,
+      use24bitFormat ? VK_FORMAT_D24_UNORM_S8_UINT : VK_FORMAT_D32_SFLOAT_S8_UINT,
       VK_FORMAT_UNDEFINED,
       0, VK_IMAGE_ASPECT_STENCIL_BIT },
     // DXGI_FORMAT_R8G8_TYPELESS
-- 
2.16.1

