diff --git a/src/d3d11/d3d11_options.cpp b/src/d3d11/d3d11_options.cpp
index 0482e2e..43aa8ea 100644
--- a/src/d3d11/d3d11_options.cpp
+++ b/src/d3d11/d3d11_options.cpp
@@ -14,7 +14,9 @@ namespace dxvk {
     
     return appOptions != g_d3d11AppOptions.end()
       ? appOptions->second
-      : D3D11OptionSet();
+      : env::getEnvVar(L"DXVK_MAP_DO_NOT_WAIT") == "1"
+        ? D3D11OptionSet(D3D11Option::AllowMapFlagNoWait)
+        : D3D11OptionSet();
   }
   
 }
\ No newline at end of file
