diff --git b/dlls/xaudio2_7/xaudio_dll.c a/dlls/xaudio2_7/xaudio_dll.c
index 28cf98565b..0682a5f3b3 100644
--- b/dlls/xaudio2_7/xaudio_dll.c
+++ a/dlls/xaudio2_7/xaudio_dll.c
@@ -89,6 +89,8 @@ BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD reason, void *pReserved)
 
     switch (reason)
     {
+    case DLL_WINE_PREATTACH:
+        return FALSE;  /* prefer native version */
     case DLL_PROCESS_ATTACH:
         instance = hinstDLL;
         DisableThreadLibraryCalls( hinstDLL );
