--- /usr/portage/app-emulation/wine-any/wine-any-3.9.ebuild	2018-05-29 08:07:05.975256001 -0300
+++ original.ebuild	2018-06-12 11:30:01.939923044 -0300
@@ -28,6 +28,8 @@
 STAGING_DIR="${WORKDIR}/${STAGING_P}"
 D3D9_P="wine-d3d9-${PV}"
 D3D9_DIR="${WORKDIR}/wine-d3d9-patches-${D3D9_P}"
+PBA_DIR="${WORKDIR}/wine-pba-${PV}"
+DOITSUJIN_DIR="${WORKDIR}/wine-doitsujin-${PV}"
 GWP_V="20180120"
 PATCHDIR="${WORKDIR}/gentoo-wine-patches"
 
@@ -36,6 +38,8 @@
 SRC_URI="${SRC_URI}
 	https://dev.gentoo.org/~np-hardass/distfiles/wine/gentoo-wine-patches-${GWP_V}.tar.xz
 "
+PBA_EGIT_REPO_URI="https://github.com/Firerat/wine-pba.git"
+DOITSUJIN_EGIT_REPO_URI="https://github.com/doitsujin/wine-hacks.git"
 
 if [[ ${PV} == "9999" ]] ; then
 	STAGING_EGIT_REPO_URI="https://github.com/wine-staging/wine-staging.git"
@@ -48,7 +52,7 @@
 
 LICENSE="LGPL-2.1"
 SLOT="${PV}"
-IUSE="+abi_x86_32 +abi_x86_64 +alsa capi cups custom-cflags d3d9 dos elibc_glibc +fontconfig +gecko gphoto2 gsm gssapi gstreamer +jpeg kerberos kernel_FreeBSD +lcms ldap +mono mp3 ncurses netapi nls odbc openal opencl +opengl osmesa oss +perl pcap pipelight +png prelink pulseaudio +realtime +run-exes s3tc samba scanner sdl selinux +ssl staging test themes +threads +truetype udev +udisks v4l vaapi vkd3d vulkan +X +xcomposite xinerama +xml"
+IUSE="+abi_x86_32 +abi_x86_64 +alsa capi cups custom-cflags d3d9 doitsujin dos elibc_glibc +fontconfig +gecko gphoto2 gsm gssapi gstreamer +jpeg kerberos kernel_FreeBSD +lcms ldap +mono mp3 native-xaudio2 ncurses netapi nls odbc openal opencl +opengl osmesa oss pba +perl pcap pipelight +png prelink pulseaudio +realtime +run-exes s3tc samba scanner sdl selinux +ssl staging test themes +threads +truetype udev +udisks v4l vaapi vkd3d vulkan +X +xcomposite xinerama +xml"
 REQUIRED_USE="|| ( abi_x86_32 abi_x86_64 )
 	X? ( truetype )
 	elibc_glibc? ( threads )
@@ -344,6 +348,16 @@
 		fi
 	fi
 
+	if use pba; then
+		git-r3_fetch "${PBA_EGIT_REPO_URI}" "${PBA_COMMIT}"
+		git-r3_checkout "${PBA_EGIT_REPO_URI}" "${PBA_DIR}"
+	fi
+
+	if use doitsujin; then
+		git-r3_fetch "${DOITSUJIN_EGIT_REPO_URI}" "${DOITSUJIN_COMMIT}"
+		git-r3_checkout "${DOITSUJIN_EGIT_REPO_URI}" "${DOITSUJIN_DIR}"
+	fi
+
 	default
 
 	l10n_find_plocales_changes "${S}/po" "" ".po"
@@ -384,6 +398,21 @@
 		fi
 		PATCHES+=( "${D3D9_DIR}/wine-d3d9.patch" )
 	fi
+	if use pba; then
+		for PATCH in ${PBA_DIR}/patches/*.patch; do
+			einfo "Applying PBA patch ${PATCH}"
+			PATCHES+=( "${PATCH}" )
+		done
+	fi
+	if use doitsujin; then
+		for PATCH in ${DOITSUJIN_DIR}/*.patch; do
+			einfo "Applying DOITSUJIN patch ${PATCH}"
+			PATCHES+=( "${PATCH}" )
+		done
+	fi
+	if use native-xaudio2; then
+		PATCHES+=( "${FILESDIR}/native-xaudio2.patch" )
+	fi
 
 	default
 	eapply_bin
